datumaro/plugins/data_formats/segment_anything/exporter.py

```diff
@@ -0,0 +1,173 @@
+
+    def get_annotation_info(self, group, img_width, img_height):
+        boxes = [a for a in group if a.type == AnnotationType.bbox]
+        polygons: List[Union[Polygon, Ellipse]] = [
+            a for a in group if a.type in self._polygon_types
+        ]
+        masks = [a for a in group if a.type == AnnotationType.mask]
+
+        anns = boxes + polygons + masks
+        leader = anno_tools.find_group_leader(anns)
+        if len(boxes) > 0:
+            bbox = anno_tools.max_bbox(boxes)
+        else:
+            bbox = anno_tools.max_bbox(anns)
+        polygons = [p.as_polygon() for p in polygons]
+
+        mask = None
+        if polygons:
+            mask = mask_tools.rles_to_mask(polygons, img_width, img_height)
+        if masks:
+            masks = (m.image for m in masks)
+            if mask is not None:
+                masks = chain(masks, [mask])
+            mask = mask_tools.merge_masks(masks)
+        if mask is None:
+            return None
+        mask = mask_tools.mask_to_rle(mask)
+
+        segmentation = {
+            "counts": list(int(c) for c in mask["counts"]),
+            "size": list(int(c) for c in mask["size"]),
+        }
+        rles = mask_utils.frPyObjects(segmentation, img_height, img_width)
+        if isinstance(rles["counts"], bytes):
+            rles["counts"] = rles["counts"].decode()
+        area = mask_utils.area(rles)
+
+        annotation_data = {
+            "id": leader.group,
+            "segmentation": rles,
+            "bbox": bbox,
+            "area": area,
+            "predicted_iou": max(ann.attributes.get("predicted_iou", 0.0) for ann in anns),
+            "stability_score": max(ann.attributes.get("stability_score", 0.0) for ann in anns),
+            "crop_box": anno_tools.max_bbox([ann.attributes.get("crop_box", []) for ann in anns]),
+            "point_coords": list(
+                set(
+                    tuple(point_coord)
+                    for ann in anns
+                    for point_coord in ann.attributes.get("point_coords", [[]])
+                )
+            ),
+        }
+        return annotation_data
+
+    def apply(self):
+        if self._extractor.media_type() and not issubclass(self._extractor.media_type(), Image):
+            raise MediaTypeError("Media type is not an image")
+
+        os.makedirs(self._save_dir, exist_ok=True)
+
+        subsets = self._extractor.subsets()
+        pbars = self._ctx.progress_reporter.split(len(subsets))
+
+        max_image_id = 1
+        for pbar, (subset_name, subset) in zip(pbars, subsets.items()):
+            for item in pbar.iter(subset, desc=f"Exporting {subset_name}"):
+                try:
+                    # make sure file_name is flat
+                    file_name = self._make_image_filename(item).replace("/", "__")
+                    try:
+                        image_id = int(item.attributes.get("id", max_image_id))
```

Comment:
can I ask you why this is `item.attributes.id` instead of `item.id`?

Reply:
`image_id` field in `segment_anything` format should be an integer.
In Datumaro, `item.id` is a string type which can be anything and is set to media file path without extension by convention.
The `segment_anything` format also follows this convention while importing so `image_id` field in the format is assigned to `item.attribute.id` instead.